modules/ROOT/pages/dataweave-type-selector.adoc

```diff
@@ -0,0 +1,209 @@
+= Type Selector
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+The type selector allows the use of the `.` selector over xref:dataweave-type-system.adoc#composite-types[types] which enables you to easily declare types from existing ones. This feature can also be combined with the new xref:dataweave-selectors-reusing-types.adoc[module loader feature] to load files and translate declarations inside your file into DataWeave type directives which can be accessed in the same way as types from any other module:
+
+Example: DataWeave script
+[source,dataweave,linenums]
+----
+%dw 2.0
+---
+type User = {name: String}
+type NameType = User.name
+var name: NameType = "Seba"
+---
+typeOf(name)
+----
+
+Output 
+----
+"String"
+----
+
+== Examples on How to Use Type Selector
+
+The following example describes how type selection is useful for declaring complex types from other types already declared:
+
+Example: Composition
+
+DataWeave script
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/json
+---
+type User = {
+ address : {country: String, city: String, street: String, number: Number},
+ userName : String
+}
+type Address = User.address
+
+var userAddress: Address = {country: "Argentina", city: "Rosario", street: "Calle Falsa", number: 123}
+---
+userAddress
+----
+
+Output
+[source,json,linenums]
+----
+{
+ "country": "Argentina",
+ "city": "Rosario",
+ "street": "Calle Falsa",
+ "number": 123
+}
+----
+
+The next example shows how type selection can also be used with Union Types involving object types:
+
+Example: Union Types
+
+DataWeave script
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/json
+---
+type ID = {name: String} | {id: Number}
+var name: ID = {name: "Seba"}
+var id: ID = {id: 38123}
+var nameString: ID.name = "Seba"
+var idNumber: ID.id = 38123
+---
+[name, id, nameString ++ " is of type " ++ typeOf(nameString), idNumber ++ " is of type " ++ typeOf(idNumber)]
+----
+
+Output
+[source,json,linenums]
+----
+[
+  {
+    "name": "Seba"
+  },
+  {
+    "id": 38123
+  },
+  "Seba is of type String",
+  "38123 is of type Number"
+]
+----
+
+However, the following script fails because the second type is not an object type and there is no way to reference it with the selector:
```

Comment:
```suggestion
However, the following script fails because the second type is not an `Object` type and there is no way to reference it with the selector:
```